src/waveform/renderers/waveformmark.h

```diff
@@ -23,31 +31,27 @@ class WaveformMark {
     WaveformMark& operator=(const WaveformMark&) = delete;
 
     // Enable swapping
-    void swap(WaveformMark& that) {
-        std::swap(m_pPointCos, that.m_pPointCos);
-        std::swap(m_properties, that.m_properties);
-        std::swap(m_iHotCue, that.m_iHotCue);
-        std::swap(m_image, that.m_image);
-    }
-    friend void swap(WaveformMark& lhs, WaveformMark& rhs) {
-        lhs.swap(rhs);
-    }
-
-    void reset(int hotCue = kDefaultHotCue);
-
-    void setup(const QString& group, const QDomNode& node,
-               const SkinContext& context,
-               const WaveformSignalColors& signalColors);
-
-    std::unique_ptr<ControlProxy> m_pPointCos;
+    // Swapping is done in the destructor
+    
 
     const WaveformMarkProperties& getProperties() const { return m_properties; };
     void setProperties(const WaveformMarkProperties& properties) { m_properties = properties; };
 
     int getHotCue() const { return m_iHotCue; };
     void setHotCue(int hotCue) { m_iHotCue = hotCue; };
 
+    //The m_pPointCos related function
+    bool is_m_pPointCos_Null() const{ return m_pPointCos == nullptr; }
+    ConfigKey getConfigKey(){ return m_pPointCos->getKey(); }
+    void changeKeyPosition(ControlObject *pObj) { m_pPointCos = std::make_unique<ControlProxy> (pObj->getKey()); }
+    void connectPlayPosChanged(const QObject *, const char *);
+    double getPlayPosition(){ return m_pPointCos->get(); }
+    
+
+    ~WaveformMark(); //set the destructor which will use the reset functionallity and do the same thing
```

Comment:
Comment does not make any sense. We don't even need to declare the destructor explicitly, because the implicitly generated destructor is sufficient.

We should instead consider to declare the class as 'final', if we don't provide a 'virtual' destructor.